src/baldr/graphtile.cc

```diff
@@ -224,15 +224,57 @@ void store(const std::string& cache_location,
 graph_tile_ptr GraphTile::CacheTileURL(const std::string& tile_url,
                                        const GraphId& graphid,
                                        tile_getter_t* tile_getter,
-                                       const std::string& cache_location) {
+                                       const std::string& cache_location,
+                                       uint64_t range_offset,
+                                       uint64_t range_size,
+                                       const std::filesystem::path& id_txt_path,
+                                       uint64_t id_checksum) {
   // Don't bother with invalid ids
   if (!graphid.Is_Valid() || graphid.level() > TileHierarchy::get_max_level() || !tile_getter) {
     return nullptr;
   }
 
-  auto fname = valhalla::baldr::GraphTile::FileSuffix(graphid.Tile_Base(),
-                                                      valhalla::baldr::SUFFIX_NON_COMPRESSED, false);
-  auto result = tile_getter->get(baldr::make_single_point_url(tile_url, fname));
+  LOG_INFO("Downloading tile " + std::to_string(graphid) + " from " + tile_url);
+
+  tile_getter_t::GET_response_t result;
+  if (range_size == 0) {
+    // requesting plain tiles
+    auto fname =
+        valhalla::baldr::GraphTile::FileSuffix(graphid.Tile_Base(),
+                                               valhalla::baldr::SUFFIX_NON_COMPRESSED, false);
+    result = tile_getter->get(baldr::make_single_point_url(tile_url, fname));
+  } else {
+    // or HTTP range on a tar
+    result = tile_getter->get(tile_url, range_offset, range_size);
```

Comment:
being pedantic: maybe it'd be better to get the tile header first which is only 272 bytes, then do the check and fail the request before downloading the whole thing only to fail right after.